processing/src/main/java/org/apache/carbondata/processing/merger/CarbonCompactionUtil.java

```diff
@@ -147,6 +147,7 @@ private static void groupCorrespodingInfoBasedOnTask(TableBlockInfo info,
         blockInfo.setDataFileFooter(dataFileMatadata);
       } else {
         dataFileMatadata = CarbonUtil.readMetadataFile(blockInfo);
+        blockInfo.setDataFileFooter(dataFileMatadata);
```

Comment:
consider to move this sentence （Line147/150） out of the branches cause all the branches have them

Reply:
done